File: src/main/java/com/baulsupp/oksocial/credentials/FixedTokenCredentialsStore.java
package com.baulsupp.oksocial.credentials;

import java.util.Optional;

public class FixedTokenCredentialsStore implements CredentialsStore {
  private String token;

  public FixedTokenCredentialsStore(String token) {
    this.token = token;
  }

  @Override public <T> Optional<T> readDefaultCredentials(ServiceDefinition<T> serviceDefinition) {
    return Optional.of(serviceDefinition.parseCredentialsString(token));
  }

  @Override
  public <T> void storeCredentials(T credentials, ServiceDefinition<T> serviceDefinition) {
    throw new UnsupportedOperationException();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/credentials/FixedTokenCredentialsStore.kt
package com.baulsupp.oksocial.credentials

class FixedTokenCredentialsStore(private val token: String) : CredentialsStore {

  override fun <T> get(serviceDefinition: ServiceDefinition<T>, tokenSet: String?): T? {
    return serviceDefinition.parseCredentialsString(token)
  }

  override fun <T> set(
    serviceDefinition: ServiceDefinition<T>, tokenSet: String?, credentials: T) {
    throw UnsupportedOperationException()
  }

  override fun <T> remove(serviceDefinition: ServiceDefinition<T>, tokenSet: String?) {
  }
}
